### PR TITLE
zscaler_zia: Fix url parsing in Web logs.

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
```
Fix url parsing in Web logs.

3.0.0 version introduced a bug in url parsing due to lack of url.scheme.
Earlier versions had logic to add url.scheme to `eurl` field to 
generate `url.full`. When the field changed from `eurl` to `url` in 3.0.0, 
this logic was removed. This leads to erroneous values in ECS `url` 
fields. This PR fixes the issue.

   - Add `network.protocol` as scheme to `url` field to fully 
     generate `url.full`.
   - If `network.protocol` is missing, add default `https` as scheme. 
     This is required to properly run `uri_parts` processor.

```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -1328,6 +1328,23 @@ processors:
...
+  - script:
+      lang: painless
+      description: Add scheme to url if not already present
+      if: ctx.zscaler_zia?.web?.url?.name != null
+      source: |-
+        def regex = /[a-z]+:\/\/.*/;
+        def url = ctx.zscaler_zia.web.url.name;
+        if (!regex.matcher(url).matches()) {
+          if (ctx.network?.protocol != null) {
+            url = ctx.network.protocol + '://' + url;
```
Comment (efd6) [inline packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml:1340]: There may be some additional work required to cover edge cases; from the [docs](https://help.zscaler.com/zia/web-insights-logs-filters#Proto), the `proto` log field (which becomes `network.protocol`) is not a direct mapping to a scheme. Though maybe that should happen in the assignment to `network.protocol`.

**Review by efd6 — APPROVED** [review]

File: packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (kcreddy) [reply packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml]: @efd6, could you please review if this seems valid:

I updated the logic to derive `network.protocol`, first based on [`alpn_protocol`](https://help.zscaler.com/zia/nss-feed-output-format-web-logs) log field as it closely matches the ECS definition. If `alpn_protocol` is absent, I added a mapping to derive the `network.protocol` for different values of `proto` log field to cover edge cases.
Changes in [8ff679f](https://github.com/elastic/integrations/pull/10517/commits/8ff679fb70cb0f6821f9ac61410495cb0859a40a).

File: packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -857,11 +857,36 @@ processors:
...
+      tag: set_network_protocol_from_web_alpn_protocol
+      copy_from: zscaler_zia.web.alpn_protocol
       ignore_empty_value: true
+  - script:
+      lang: painless
+      if: ctx.network?.protocol == null
+      description: Set network.protocol from prototype, if not already set.
+      tag: script_set_network_protocol
+      params:
+        dns-over-https: dns
```
Comment (efd6) [inline packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml:872]: Worth having a source link for the origin of this table, though I can't find a good source that includes all the protocols that are here, so if it doesn't exist, never mind.

**Review by efd6 — APPROVED** [review]

File: packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (kcreddy) [reply packages/zscaler_zia/data_stream/web/elasticsearch/ingest_pipeline/default.yml:872]: I also couldn't find a mapping table, but I will add link to the list of uri schemes

**Pull request merged**